apps/web/src/components/WalletConnect/SessionProposalModal.tsx

```diff
@@ -0,0 +1,218 @@
+      try {
+        await walletKit.rejectSession({
+          id: proposal.id,
+          reason: getSdkError("USER_REJECTED_METHODS"),
+        });
+        onClose();
+      } catch (e) {
+        toast({ description: (e as Error).message, status: "error" });
+        setIsLoadingReject(false);
+        return;
+      }
+    });
+
+  return (
+    <FormProvider {...form}>
+      <ModalContent>
+        <RequestModalContainer title="">
+          {!isValid && (
+            <Box marginBottom="16px" textAlign="left">
+              <Card marginLeft="8px">
+                Fill in the account. Then if needed inspect the console for more information.
+              </Card>
+            </Box>
+          )}
+          <ProjectInfoCard metadata={proposal.params.proposer.metadata} />
+          <Divider />
+          <Box marginBottom="16px" fontSize="xl" fontWeight="semibold">
+            Requested permissions
+          </Box>
+
+          <VStack align="start" spacing="8px">
+            <HStack>
+              <Icon as={CheckmarkIcon} />
+              <Card marginLeft="8px">View your balance and activity</Card>
+            </HStack>
+            <HStack>
+              <Icon as={CheckmarkIcon} />
+              <Card marginLeft="8px">Send approval requests</Card>
+            </HStack>
+            <HStack color="gray.500">
+              <Icon as={CloseIcon} />
+              <Card marginLeft="8px">Move funds without permission</Card>
+            </HStack>
+          </VStack>
+
+          <Box marginTop="8px">
+            {network ? (
+              <>
+                <FormControl marginTop="24px" isInvalid={!!errors.address}>
+                  <OwnedImplicitAccountsAutocomplete
+                    allowUnknown={false}
+                    inputName="address"
+                    label="Select Account"
+                  />
+                  {errors.address && <FormErrorMessage>{errors.address.message}</FormErrorMessage>}
+                </FormControl>
+                <Text marginTop="16px" color="gray.500">
+                  Network:
+                </Text>
+                <Text marginLeft="8px">{network}</Text>
+              </>
+            ) : (
+              <>
+                <Text color="gray.600">Accounts</Text>
+                <Text>None available</Text>
+                <Text marginTop="16px" color="gray.500">
+                  Network
+                </Text>
+                <Text>None of the required networks is supported</Text>
+              </>
+            )}
+          </Box>
+          <Divider />
+          <VerifyInfobox />
+        </RequestModalContainer>
+        <ModalFooter>
+          <Button
+            width="100%"
+            isDisabled={false}
+            isLoading={isLoadingReject}
+            onClick={onReject}
+            size="lg"
+          >
+            Reject
+          </Button>
+          <Button
+            width="100%"
+            isDisabled={!!error || !isValid}
```

Comment:
just `error || !isValid` is sufficient

Reply:
`error` is a `string`. 
`!error` - bool, true for undefined or empty error
`!!error` - bool, true for defined error

I get compilation error if change to `error`